python/lsst/ts/m2gui/controltab/tab_utility_view.py

```diff
@@ -33,7 +39,424 @@ class TabUtilityView(TabDefault):
+            "power_current_motor": create_label(),
+            "power_voltage_communication": create_label(),
+            "power_current_communication": create_label(),
+            "inclinometer": create_label(),
+        }
+
+        self._breakers = self._create_indicators_breaker()
+        self._button_reset_breakers = set_button(
+            "Reset Breakers", self._callback_reset_breakers
+        )
+
+        self._temperatures = self._create_labels_sensor_data(
+            self.model.utility_monitor.temperatures
+        )
+
+        self._displacements = self._create_labels_sensor_data(
+            self.model.utility_monitor.displacements
+        )
+
+        self._set_widget_and_layout()
+
+        self._set_signal_utility(self.model.utility_monitor.signal_utility)
+
+    def _create_indicators_breaker(self):
+        """Create the indicators of breakers.
+
+        Returns
+        -------
+        indicators : `dict [PySide2.QtWidgets.QPushButton]`
+            Indicators of the breakers. The key is the breaker's name.
+        """
+
+        indicators = dict()
+        for name in self.model.utility_monitor.breakers.keys():
+            indicators[name] = set_button(
+                name, None, is_indicator=True, is_adjust_size=True
+            )
+            self._update_indicator_color(indicators[name], False)
+
+        return indicators
+
+    def _update_indicator_color(self, indicator, is_fault):
+        """Update the color of indicator.
+
+        Parameters
+        ----------
+        indicator : `PySide2.QtWidgets.QPushButton`
+            Indicator.
+        is_fault : `bool`
+            Is in fault or not.
+        """
+
+        palette = indicator.palette()
+
+        if is_fault:
+            palette.setColor(QPalette.Button, Qt.red)
+        else:
+            palette.setColor(QPalette.Button, Qt.green)
+
+        indicator.setPalette(palette)
+
+    @Slot()
+    def _callback_reset_breakers(self):
+        """Callback of the reset-breakers button. This will reset all triggered
+        breakers."""
+        self.model.utility_monitor.reset_breakers()
+
+    def _create_labels_sensor_data(self, sensor_data):
+        """Create the labels of sensor data.
+
+        Parameters
+        ----------
+        sensor_data : `dict`
+            Sensor data. The key is the sensor's name.
+
+        Returns
+        -------
+        labels_sensor_data : `dict [PySide2.QtWidgets.QLabel]`
+            Labels of the sensor data. The key is the sensor's name.
+        """
+
+        labels_sensor_data = dict()
+        for sensor in sensor_data.keys():
+            labels_sensor_data[sensor] = create_label()
+
+        return labels_sensor_data
+
+    def _set_widget_and_layout(self):
+        """Set the widget and layout."""
+
+        widget = self.widget()
+        widget.setLayout(self._create_layout())
+        self.setWidget(self.set_widget_scrollable(widget, is_resizable=True))
```

Comment:
Why do you need widget scrolable? Starting the GUI, I received:

![image](https://user-images.githubusercontent.com/5473542/166443813-00e1d4d3-701e-43a0-9652-16dfe5b69721.png)
Removing scrollable, a much better image appears:
![image](https://user-images.githubusercontent.com/5473542/166443993-7a9bbd34-dfe8-495c-908b-84abdae9afed.png)

Size is dynamically updated. you shouldn't use scrollable unless needed e.g. for a text/big graphics.

Reply:
Can you please at least make window by default big enough to hold all values without scrolling if possible? It's quite annoying to lunch window just to have it resize. You can also store window size in config, see ts_cRIOpy fo details:
```python
    @asyncClose
    async def closeEvent(self, event):
        settings = QSettings("LSST.TS", "M1M3GUI")
        settings.setValue("geometry", self.saveGeometry())
        settings.setValue("windowState", self.saveState())
        await asyncio.gather(
            self.m1m3.close(),
            self.mtmount.close(),
            self.compressor_1.close(),
            self.compressor_2.close(),
        )
        super().closeEvent(event)
```
this will make operator life a bit easier.

Reply:
I tried the following code but it can not work:

```
    def _set_widget_and_layout(self):
        """Set the widget and layout."""

        layout = self._create_layout()

        widget = self.widget()
        widget.setLayout(layout)

        scroll_area = self.set_widget_scrollable(widget, is_resizable=True)

        # scroll_area.setMinimumSize(layout.minimumSize())
        scroll_area.resize(layout.minimumSize())
        scroll_area.adjustSize()

        self.setWidget(scroll_area)
```

Only the `scroll_area.setMinimumSize()` can adjust the size of viewport. However, this is not what I am looking for. Is there any idea? Thanks!

Reply:
You need to resize dock to hintSize of the widget (obviously; you need to resize your window, not scroll area):
```
diff --git a/python/lsst/ts/m2gui/controltab/tab_config_view.py b/python/lsst/ts/m2gui/controltab/tab_config_view.py
index cd4e5cb..7b0bfe9 100644
--- a/python/lsst/ts/m2gui/controltab/tab_config_view.py
+++ b/python/lsst/ts/m2gui/controltab/tab_config_view.py
@@ -79,6 +79,9 @@ class TabConfigView(TabDefault):
 
         widget = self.widget()
         widget.setLayout(self._create_layout())
+
+        self.resize(widget.sizeHint())
+
         self.setWidget(self.set_widget_scrollable(widget, is_resizable=True))
 
     def _create_layout(self):
```

Reply:
Fixed in the commit: [2927fa9](https://github.com/lsst-ts/ts_m2gui/pull/7/commits/2927fa94e6b229312b4685c13217911ee3dca4f0). Thanks!